File: sdk/ruby/examples/hello.rb
# Example FunctionFly function in Ruby.
# Build: mruby main.rb -o hello.wasm (or use the FunctionFly bundler)
require_relative "../lib/functionfly"

class HelloFunction < FunctionFly::Function
  def handle(input, ctx)
    ctx.log("Hello from Ruby function!")
    '{"message": "Hello from Ruby!", "ok": true}'
  end
end

FunctionFly.run(HelloFunction)

File: sdk/rust/examples/hello.rs
use functionfly_sdk::{Function, Context, run};

struct HelloFunction;

impl Function for HelloFunction {
    fn handle(&self, _input: &str, _ctx: &Context) -> Result<String, String> {
        functionfly_sdk::host_functions::log("Hello from Rust function!");
        Ok(r#"{"message": "Hello from Rust!", "ok": true}"#.to_string())
    }
}

run!(HelloFunction);
